### PR TITLE
Session management

### PR DESCRIPTION
**Pull request by stevenvegt**
Adds session management using a jwt.
Create a session key pair during startup.
Accepts a Bearer token in the `Authorization` request header.
Token expires after 20 minutes.
Logs username in the console.
Clients can handle simple exceptions.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -42,13 +75,27 @@ func main() {
...
+			Password string
+		}{}
+		err := ctx.Bind(&credentials)
+		if err != nil {
+			return err
+		}
+		if credentials.Username != config.Credentials.Username|| credentials.Password != config.Credentials.Password {
+			return echo.NewHTTPError(http.StatusForbidden, "invalid credentials")
+		}
+		return ctx.JSON(200, map[string]string{"token": "123"})
```
Comment (reinkrul) [inline main.go:90]: if we keep all state client-side (put it all in the token) we probably don't need a server-side session?

**Review by reinkrul — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (stevenvegt) [reply main.go]: When we are planning to open up our demo apps coupled to the development network to the public interwebs we need  real auth. So we do need some form of server sessions.

File: main.go [hunk 2]
```diff
@@ -30,8 +43,78 @@ func getFileSystem(useFS bool) http.FileSystem {
...
+		log.Printf("unable to parse token: %s", err)
+		return nil, err
+	}
+	return t, nil
+}
+
+var sessionKey crypto.PrivateKey
+
+func generateSessionKey() (crypto.PrivateKey, error) {
+	key, err := rsa.GenerateKey(rand.Reader, 2048)
```
Comment (reinkrul) [inline main.go:102]: lets use ecdsa (P-256) for quicker startup and smaller keys? (sensible default)

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (reinkrul) [reply main.go]: I don't say we don't need auth (we do need, as you correctly point out), just saying that we shouldn't introduce sessions (in near future) when we don't need them

**Review by reinkrul — APPROVED** [review]

**Pull request merged**